### PR TITLE
chore: upgrade react-hook-form to 7.21.2

### PR DESCRIPTION
**Pull request by cjthompson**
## Description

chore: upgrade react-hook-form to 7.20.1

Cleanup Send modal:
* Flatten "crypto" and "fiat"
* Fix typing of "useWatch" calls
* Check for undefined form values

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [X] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [X] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [X] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

### CODE REVIEW

**Review by elmutt — APPROVED** [review]

**Review by reidrankin — APPROVED** [review]

**Comment by 0xdef1cafe** [comment]
i vaguely recall losing a bunch of type information on `useWatch` hook calls with this version or otherwise, we may want to check with @theoboldfrazier as he used that extensively in yearn stuff.

**Comment by cjthompson** [comment]
I did a quick test of this on CloudFlare and it seemed to work. Do we want to merge this or get more testing done?

File: src/components/Modals/Send/views/Address.tsx [hunk 1]
```diff
@@ -20,20 +20,25 @@ import { useChainAdapters } from 'context/ChainAdaptersProvider/ChainAdaptersPro
...
 export const Address = () => {
   const history = useHistory()
   const translate = useTranslate()
   const {
     formState: { errors }
-  } = useFormContext()
-  const [address, asset] = useWatch({ name: [SendFormFields.Address, SendFormFields.Asset] })
+  } = useFormContext<SendInput>()
+  const address = useWatch<SendInput, SendFormFields.Address>({ name: SendFormFields.Address })
+  const asset = useWatch<SendInput, SendFormFields.Asset>({ name: SendFormFields.Asset })
```
Comment (cjthompson) [inline src/components/Modals/Send/views/Address.tsx:33]: using `name` as an array has problems with typing, the values are typed as `never`. After some research it looks like not using the array syntax for `name` is a reasonable workaround. The other option is just to watch the entire form.

File: src/components/Modals/Send/views/Address.tsx [hunk 2]
```diff
@@ -20,20 +20,25 @@ import { useChainAdapters } from 'context/ChainAdaptersProvider/ChainAdaptersPro
...
+
   const chainAdapters = useChainAdapters()
-  const adapter = chainAdapters.byChain(asset.chain)
   const { send } = useModal()
 
+  if (!(asset?.chain && asset?.name && address)) return null
```
Comment (cjthompson) [inline src/components/Modals/Send/views/Address.tsx:38]: I don't know if returning `null` is the right thing here vs. some form of placeholder, but I'm no UI dev so I'd prefer to leave that up to someone else.

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by 0xdef1cafe** [comment]
thank you for this i tried this once and pulled my hair out

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**